appscale/agents/base_agent.py

```diff
@@ -249,12 +249,7 @@ def diff(self, list1, list2):
     Returns:
       A list of elements unique to list1
     """
-    diffed_list = []
-    list2 = set(list2)
-    for item in list1:
-      if item not in list2:
-        diffed_list.append(item)
-    return diffed_list
+    return list(set(list1).difference(set(list2)))
```

Comment:
Not sure if it's important, but the previous method preserved the order of items from `list1`.